### DISCUSSION TITLE
Connecting sensors

### DISCUSSION THREAD

**Discussion by hitchclimber** [japaric/discovery#503] [Q&A] [answered]
How do I know which pins to use to connect arbitrary peripherals? I couldn't find anything in the user manual.

**Comment by sirhcel** [comment, marked as answer]
You will find this information typically in the device's datasheet. The reference manual typically covers a whole family of devices and therefor this information is given in the data sheets which are more specific to individual variants.

For example for the [STM32F303](https://www.st.com/en/microcontrollers-microprocessors/stm32f303vc.html#documentation), table 13, _STM32F303xB/STM32F303xC pin definitions_ gives an overview on which peripheral (UART, I2C, ...) you could connect to which pin. 

From today's perspective, the STM32F303 is a somewhat older microcontroller where you could connect a function to a small number of different pin sets. Today's controllers often provide more flexiblity. They often allow to connect arbitrary functions and pins.

**Reply by hitchclimber** [reply]
Thanks for the swift and thorough reply!
